python/hail/dataset.py

```diff
@@ -3714,6 +3720,52 @@ def storage_level(self):
         return self._jvds.storageLevel()
 
     @handle_py4j
+    @requireTGenotype
+    def summarize(self):
+        """Returns a summary of useful information about the dataset.
+        
+        .. include:: requireTGenotype.rst
+
+        
+        **Examples**
+        
+        >>> s = vds.summarize()
+        >>> print(s.contigs)
+        >>> print('call rate is %.2f' % s.call_rate)
+        >>> s.report()
+        
+        The following information is contained in the summary:
+        
+         - **samples** (*int*) - Number of samples.
+         - **variants** (*int*) - Number of variants.
+         - **call_rate** (*float*) - Fraction of all genotypes called.
+         - **contigs** (*list of str*) - List of all unique contigs found in the dataset.
+         - **multiallelics** (*int*) - Number of multiallelic variants.
+         - **snps** (*int*) - Number of SNP alternate alleles.
+         - **indels** (*int*) - Number of insertion / deletion alternate alleles.
+         - **complex** (*int*) - Number of complex (star, MNP, etc) alternate alleles.
+         - **most_alleles** (*int*) - The highest number of alleles at any variant.
+         
+        :return: Object containing summary information.
+        :rtype: :class:`~hail.utils.Summary`
+        """
+
+        js = self._jvdf.summarize()
+        return Summary(
```

Comment:
I think it might be cleaner to have a python `Summary` constructor that takes an instance of the Scala object, then get all the fields out of the Scala object within that constructor, as opposed to deconstructing the Scala object here.